src/components/MasterItemsSelect.js

```diff
@@ -1,25 +1,61 @@
 import { Component, PropTypes, DOM as dom } from 'react';
+import _ from 'underscore';
 
 class MasterItemsSelect extends Component {
   constructor(props) {
     super(props);
     this.handleChange = this.handleChange.bind(this);
+    this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    this.state = {
+      filteredItems: this.filterItems(this.props.masterItems[0].category)
+    };
   };
 
+  getCategories(masterItems) {
+    return masterItems.map((item) => item.category);
+  }
+
+  filterItems(category) {
+    return this.props.masterItems.filter((item) => item.category === category);
+  }
+
   handleChange(e) {
     const itemId = e.target.value;
-    this.props.onSelectMasterItem(itemId);
+    const selectedItem = _.find(this.props.masterItems, (item) => item.id === itemId);
+    this.props.onSelectMasterItem(selectedItem);
+  }
+
+  handleChangeCategory(e) {
+    const selectedCategory = e.target.value;
+    const filteredItems = this.filterItems(selectedCategory);;
+
+    this.props.onSelectMasterItem(filteredItems[0]);
```

Comment:
This also feels weird to me...had to set the selected item to the first in the category everytime the category changes.